src/drivers/esp8266_drv.c

```diff
@@ -1408,8 +1426,11 @@ static void connect_cb(const bool status, const bool already_connected)
 				"connected: ", cso->chan_id);
                 channel_close(ch);
         } else if (!status) {
-		pr_info_int_msg(LOG_PFX "Failed to connect channel: ",
+                pr_info_int_msg(LOG_PFX "Failed to connect channel: ",
 				cso->chan_id);
+                esp8266_state.cmd.failures++;
```

Comment:
Here, we were not counting connect failures as errors. 

Sometimes our esp8266 does a hard reset, which resets the baud rate, causing future connections to always fail. this will catch it and eventually cause a hard reset.

Reply:
Oh fun.  Stupid ESP8266.